filters/include/pcl/filters/experimental/grid_filter.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/filters/filter.h>
+
+#include <cfloat> // for FLT_MAX
+
+namespace pcl {
+namespace experimental {
+
+template <typename T>
+using get_point_type = typename T::PointCloud::PointType;
+
+template <typename T>
+using get_grid_type = typename T::Grid;
+
+template <typename GridStruct,
```

Comment:
Do you mean some counterpart like `FilterIndices`? But do some transform (e.g. random sampling) instead of binary removal?

Reply:
> But then this kind of transform filters don't have common logic/method basically

No common logic/method needed. Just bare bone functionality. The common API will be provided by `GridFilterBase`

Reply:
What would be included in `TransformAndFilter`? Like the logic inside `applyFilter`?

Reply:
applyFilter and the absolute minimum required stuff

Reply:
I would prefer not splitting for now. 
- The API in `GridFilter` now is already quite minimum
- We have to also consider other non binary removal filters, since in principle they also belong to `TransformAndFilter`

We can come back to this after finishing the grid filters